docs/src/userguide/config/event_config.rst

```diff
@@ -126,13 +126,176 @@ Example configuration for a custom event input that is not supported by Klio:
 
     Any arbitrary key-value pairs for custom event input configuration specific to a job.
 
+.. _event-config-bigquery:
 
-.. todo:: document supported batch event inputs
+GoogleBigQuery
+^^^^^^^^^^^^^^
+
+*Mode: Batch*
+
+Consuming KlioMessages from `Google BigQuery`_ is supported for all runners. Klio allows the user to specify events from queries and from tables.
+
+Example configuration for `Google BigQuery`_, by specifying a table:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              columns:
+              - entity_id
+              dataset: bigquery_dataset
+              project: gcp_project
+              table: bigquery_table
 
 
+Example configuration for `Google BigQuery`_, by specifying a query in the BigQuery dialect:
+
+    .. code-block:: yaml
+
+        name: my-cool-job
+        pipeline_options:
+          streaming: True
+        job_config:
+        events:
+            inputs:
+            - type: bq
+              query:
+                SELECT * FROM [gcp_project.bigquery_dataset.bigquery_table]
+
+.. option:: job_config.events.inputs[].type
+
+    Value: ``bq``
+
+    | **Runner**: Dataflow, Direct
+    | *Required*
+
+.. option:: job_config.events.inputs[].columns[]
+
+    Not applicable if query is specified.
+
+    A list of strings specifying the columns to read events from.
+    If only one column is provided, the value will be returned as a bytestring.
+    If no columns are specified, or if more than one column is specified,
+    the results including the column names will be serialized to a JSON bytestring,
+    e.g. ``'{"field1": "foo", "field2": bar"}'``.
+
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].columns[].<column> STR
+
+    A column in the table or query result used to build the event input from.
+
+    | **Runner**: Dataflow, Direct
+    | *Optional*
+
+.. option:: job_config.events.inputs[].table STR
+
+    Name of the event input table.
+
+    If specified all data of the table will be used as input of the current source.
```

Comment:
"If specified, all..." (+comma)

Also, I'm not sure what this means. If a table is given, **all** data is used? or does it mean, "If no `column` names are listed in `job_config.events.inputs[].columns[]`, then all table data is used."

Reply:
I just removed it - not sure what I was going for here and the previous sentence expresses the purpose pretty well.